frontend/common/src/components/SkillPicker/SkillBlock.tsx

```diff
@@ -0,0 +1,181 @@
+    data-h2-font-weight="base(700)"
+  >
+    <CheckCircleIcon data-h2-width="base(x.75)" />
+    <span>{children}</span>
+  </span>
+);
+
+interface SkillLabelProps {
+  children: React.ReactNode;
+  isAdded: boolean;
+}
+
+const SkillLabel = ({ children, isAdded }: SkillLabelProps) => {
+  const Wrapper = isAdded ? SkillLabelWrapper : React.Fragment;
+
+  return (
+    <Wrapper>
+      <span>{children}</span>
+    </Wrapper>
+  );
+};
+
+interface SkillBlockProps {
+  skill: Skill;
+  isAdded: boolean;
+  onAddSkill: (id: string) => void;
+  onRemoveSkill: (id: string) => void;
+}
+
+const SkillBlock = ({
+  skill: { id, name, description },
+  isAdded,
+  onAddSkill,
+  onRemoveSkill,
+}: SkillBlockProps) => {
+  const intl = useIntl();
+  const locale = useLocale();
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+  const definition = description ? description[locale] : null;
+  const skillName = getLocalizedName(name, intl);
+
+  const Wrapper = definition ? Collapsible.Root : "div";
+
+  return (
+    <Wrapper>
+      <div
+        data-h2-display="base(flex)"
+        data-h2-flex-direction="base(column) p-tablet(row)"
+        data-h2-align-items="base(flex-start) p-tablet(center)"
+        data-h2-justify-content="base(space-between)"
+        data-h2-gap="base(0, x.25) p-tablet(x.5, 0)"
+      >
+        <SkillLabel isAdded={isAdded}>{skillName}</SkillLabel>
+        <div
+          data-h2-display="base(flex)"
+          data-h2-justify-content="base(flex-end)"
+          data-h2-gap="base(x.5, 0)"
+          style={{ flexShrink: 0 }}
+        >
+          <div>
+            <Button
+              color="primary"
+              mode="inline"
+              type="button"
+              onClick={isAdded ? () => onRemoveSkill(id) : () => onAddSkill(id)}
+            >
+              {isAdded
+                ? intl.formatMessage(
+                    {
+                      defaultMessage:
+                        "Remove skill<hidden> {skillName}</hidden>",
+                      id: "p5GbCr",
+                      description:
+                        "Button label to remove skill on skill result block.",
+                    },
+                    {
+                      skillName,
+                    },
+                  )
+                : intl.formatMessage(
+                    {
+                      defaultMessage: "Add skill<hidden> {skillName}</hidden>",
+                      id: "HB2yOT",
+                      description:
+                        "Button label to add skill on skill result block.",
+                    },
+                    { skillName },
+                  )}
+            </Button>
+          </div>
+          <div>
+            {definition ? (
+              <Collapsible.Trigger asChild>
+                <Button
+                  color="primary"
+                  mode="inline"
+                  type="button"
+                  onClick={() => setIsOpen(!isOpen)}
```

Comment:
When a useState setter uses its own value, I think its best practice to pass the current value as an argument:
https://reactjs.org/docs/hooks-reference.html#functional-updates

```suggestion
                  onClick={(currentIsOpen) => setIsOpen(!currentIsOpen)}
```